src/sources/vector/mod.rs

```diff
@@ -50,6 +54,21 @@ impl proto::Service for Service {
             .map(Event::from)
             .collect();
 
+        if self.log_namespace == LogNamespace::Vector {
+            let now = Utc::now();
+            for event in &mut events {
+                if let Event::Log(ref mut log) = event {
+                    log.metadata_mut()
+                        .value_mut()
+                        .insert(path!("vector", "source_type"), VectorConfig::NAME);
```

Comment:
I had originally done that but the reason I didn't use that function here is because it changes the log in the `Legacy` namespace case, which I don't think we want to do here because this source wasn't adding these fields.

But, this could be a misconception on my part.

Reply:
🤔 is this another case of we should have been including fields and we should have a bug fix?

Reply:
Hmmm. Good question. I assumed this source was maybe under special circumstances but yeah, if they're all supposed to be having those fields then, I can swap to use the helper function and add a breaking changes note

Reply:
> it changes the log in the `Legacy` namespace case

It wouldn't because it is behind the `if self.log_namespace == LogNamespace::Vector`. But I can see how it would still cause confusion since that isn't totally obvious. I'm happy to change my mind here.

Reply:
> It wouldn't because it is behind the if self.log_namespace == LogNamespace::Vector. But I can see how it would still cause confusion since that isn't totally obvious. I'm happy to change my mind here.

Oh of course! When I first wrote this and looked at `insert_vector_metadata()` , it was before I added the if condition for the Vector namespace.
I agree with using `insert_vector_metadata` , will change to use that.

Reply:
However- just to be clear- I think this 

>  All sources adding the source_type is definitely just a "my understanding" side of things, IDK if anyone is around this week that can actually confirm though.

> Good followup question for @jszwedko when he's back.

Still merits discussion so I'll hold off on merging this and bring it up next week.

Reply:
We discussed as a team and decided adding the `source_type` to the legacy namespace for sources that are missing it is the way to go- and mentioning it in the release notes.